File: subanu/18.09.2025/constructor.dart
class Student {
  String? name;
  int? age;

  //parameterized constructor
  Student(this.name, this.age);

  //named constructor
  Student.cons1(this.name) {
    print("Named constructor 1");
  }

  Student.cons2(int age) {
    print("Age: $age in named constructor 2");
  }

  void display() {
    print("Hi");
  }
}

void main() {
  Student obj = Student("Subanu", 21);
  print(obj.name);

  Student obj1 = Student.cons1("subanu");
  Student obj2 = Student.cons2(21);
  print(obj2.age); //null
  print(obj1.name);
  obj2.display();
}

File: darshan/week-1/08-19-25 java/3-constructor-types/Constructor.java
class Student{

	String name;
	int age;

	Student(){
		//DEFAULT CONSTRUCTOR
		
		System.out.println("=== In default constructor ===");

		this.name = "default value";
		this.age = 10;
		System.out.println(name);
		System.out.println(age);

	} 

	Student(String name, int age){

		//PARAMETERISED CONSTRUCTOR
		System.out.println("=== In parameterised constructor ===");
		System.out.println(name);
		System.out.println(age);
	}

}
class Constructor{

	public static void main(String args[]){
		
		//Invokes default constructor.
		Student studentDefault = new Student();				
		
		//Invokes parameterised constructor.
		Student studentParameterised = new Student("Darshan", 23);	
		
		

	}
}